aws/resource_aws_transfer_server.go

```diff
@@ -34,6 +35,40 @@ func resourceAwsTransferServer() *schema.Resource {
 				Computed: true,
 			},
 
+			"endpoint_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  transfer.EndpointTypePublic,
+				ValidateFunc: validation.StringInSlice([]string{
+					transfer.EndpointTypePublic,
+					transfer.EndpointTypeVpcEndpoint,
+				}, false),
+			},
+
+			"endpoint_details": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vpc_endpoint_id": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
Nit: This can be simplified with the below but certainly not worth holding up this pull request for:

```go
ValidateFunc: validation.StringMatch(regexp.MustCompile(`^vpce-[0-9a-f]{17}$`), "must match VPC Endpoint identifier beginning with vpce-"),
```